src/analysis/normed_space/star/multiplier.lean

```diff
@@ -0,0 +1,414 @@
+/-
+Copyright (c) 2022 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux, Jon Bannon
+-/
+
+import algebra.star.star_alg_hom
+import analysis.normed_space.star.basic
+import analysis.normed_space.operator_norm
+import analysis.special_functions.pow
+
+/-!
+# Multiplier Algebra of a C⋆-algebra
+
+Define the multiplier algebra of a C⋆-algebra as the algebra (over `𝕜`) of double centralizers,
+for which we provide the localized notation `𝓜(𝕜, A)`.  A double centralizer is a pair of
+continuous linear maps `L R : A →L[𝕜] A` satisfying the intertwining condition `R x * y = x * L y`.
+
+There is a natural embedding `A → 𝓜(𝕜, A)` which sends `a : A` to the continuous linear maps
+`L R : A →L[𝕜] A` given by left and right multiplication by `a`, and we provide this map as a
+coercion.
+
+The multiplier algebra corresponds to a non-commutative Stone–Čech compactification in the sense
+that when the algebra `A` is commutative, it can be identified with `C₀(X, ℂ)` for some locally
+compact Hausdorff space `X`, and in that case `𝓜(𝕜, A)` can be identified with `C(β X, ℂ)`.
+
+## Implementation notes
+
+## TODO
+
++ define a type synonym for `𝓜(𝕜, A)` which is equipped with the strict uniform space structure
+  and show it is complete
++ show that the image of `A` in `𝓜(𝕜, A)` is an essential ideal
++ prove the universal property of `𝓜(𝕜, A)`
+* Construct a double centralizer from a pair of maps `L : A → A`, `R : A → A` satisfying the
+  centrality condition `∀ x y, R x * y = x * L y`.
+-/
+
+noncomputable theory
+
+open_locale nnreal ennreal
+open nnreal continuous_linear_map
+
+universes u v
+
+/-- The type of *double centralizers*, also known as the *multiplier algebra* and denoted by
+`𝓜(𝕜, A)`, of a non-unital normed algebra. -/
+@[ext]
+structure double_centralizer (𝕜 : Type u) (A : Type v) [nontrivially_normed_field 𝕜]
+  [non_unital_normed_ring A] [normed_space 𝕜 A] [smul_comm_class 𝕜 A A] [is_scalar_tower 𝕜 A A] :=
+(left : A →L[𝕜] A)
+(right : A →L[𝕜] A)
+(central : ∀ x y : A, right x * y = x * left y)
```

Comment:
In fact, instead of defining `prod_mk`, `to_prod`, you can niftily ectend ` (A →L[𝕜] A) × (A →L[𝕜] A)` in the first place.
```suggestion
structure double_centralizer (𝕜 : Type u) (A : Type v) [nontrivially_normed_field 𝕜]
  [non_unital_normed_ring A] [normed_space 𝕜 A] [smul_comm_class 𝕜 A A] [is_scalar_tower 𝕜 A A]
  extends (A →L[𝕜] A) × (A →L[𝕜] A) :=
(central : ∀ x y : A, snd x * y = x * fst y)
```
This is an uncommon pattern, but quite a useful one IMO.

Reply:
nice idea, I have implemented this.